langchain/src/chains/graph_qa/cypher.ts

```diff
@@ -0,0 +1,200 @@
+import { LLMChain } from "../../chains/llm_chain.js";
+import { ChainValues } from "../../schema/index.js";
+import { BasePromptTemplate } from "../../prompts/base.js";
+import { BaseChain, ChainInputs } from "../base.js";
+import { BaseLanguageModel } from "../../base_language/index.js";
+import { CallbackManagerForChainRun } from "../../callbacks/manager.js";
+import { Neo4jGraph } from "../../graphs/neo4j_graph.js";
+import { CYPHER_GENERATION_PROMPT, CYPHER_QA_PROMPT } from "./prompts.js";
+
+export const INTERMEDIATE_STEPS_KEY = "intermediateSteps";
+
+export interface GraphCypherQAChainInput extends ChainInputs {
+  graph: Neo4jGraph;
+  cypherGenerationChain: LLMChain;
+  qaChain: LLMChain;
+  inputKey?: string;
+  outputKey?: string;
+  topK?: number;
+  returnIntermediateSteps?: boolean;
+  returnDirect?: boolean;
+}
+
+export interface FromLLMInput {
+  graph: Neo4jGraph;
+  llm?: BaseLanguageModel;
+  cypherLLM?: BaseLanguageModel;
+  qaLLM?: BaseLanguageModel;
+  qaPrompt?: BasePromptTemplate;
+  cypherPrompt?: BasePromptTemplate;
+  returnIntermediateSteps?: boolean;
+  returnDirect?: boolean;
+}
+
+export class GraphCypherQAChain extends BaseChain {
```

Comment:
Would be nice to show this in expression language as well at some point if possible!